api/src/wfl/api/handlers.clj

```diff
@@ -18,7 +18,12 @@
   [body]
   (log/warn "Endpoint returned 400 failure response:")
   (log/warn body)
-  (-> body response/bad-request (response/content-type "application/json")))
+  (-> body
+      (.getMessage)
+      ;(Throwable->map)
+      ;(dissoc :trace :via :data)
```

Comment:
```clojure
(Throwable->map)
(dissoc :trace :via :data)
```
returns customizable map (through `dissoc`):
```
{
    "cause": "clj-http: status 401"
}
```
while `(.getMessage)` returns the message:
```
Failed to create workload - no such pipeline
```
Since this is just a short-term fix, I figured we don't want to take the risk of let some sensible info slip through `dissoc`.

Reply:
If the right thing to do is throw, then why wrap at all? Delete the calls to fail-on-error and let the endpoints throw?

Reply:
I believe the right thing is to throw at handler level and rely on the top level exception middleware to catch them, if you are ok with that, I can remove the `fail-on-error` calls and let the endpoints throw.

Reply:
That sounds right to me too.